feat: OPTIC-2: Save draft on leave

robot-ci-heartex opened the pull request:
Hi @Travis1282!

This PR was created in response to a PRs in upstream repo:
- https://github.com/HumanSignal/dm2/pull/231
- https://github.com/HumanSignal/label-studio-frontend/pull/1552

Travis1282 committed:
[submodules] Build static HumanSignal/dm2

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6018306785

Travis1282 committed:
[submodules] Build static HumanSignal/dm2

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6041221119

Travis1282 committed:
Merge branch 'origin/develop' into 'fb-optic-2/save-draft-on-leave'

Travis1282 committed:
[submodules] Build static HumanSignal/label-studio-frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6042978607

Travis1282 committed:
[submodules] Build static HumanSignal/dm2

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6054035666

Travis1282 committed:
Merge branch 'origin/develop' into 'fb-optic-2/save-draft-on-leave'

Travis1282 committed:
[submodules] Build static HumanSignal/label-studio-frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6054150497

Travis1282 committed:
[submodules] Build static HumanSignal/dm2

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6054211352

Travis1282 committed:
[submodules] Build static HumanSignal/dm2

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6054212296

Travis1282 committed:
[submodules] Build static HumanSignal/dm2

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6088235972

Travis1282 committed:
Merge branch 'origin/develop' into 'fb-optic-2/save-draft-on-leave'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6088235972

Travis1282 committed:
[submodules] Build static HumanSignal/label-studio-frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6088239538

Travis1282 committed:
draft guard added for preserving drafts on navagate away

Travis1282 committed:
Merge branch 'fb-optic-2/save-draft-on-leave' of https://github.com/heartexlabs/label-studio into fb-optic-2/save-draft-on-leave

bmartel commented on label_studio/frontend/src/components/DraftGuard/DraftGuard.js:
```diff
@@ -0,0 +1,55 @@
+  const history = useHistory();
+
+  useEffect(async () => {
+    if (nextLocation, isFF(FF_OPTIC_7)) {
```
```suggestion
    if (nextLocation && isFF(FF_OPTIC_7)) {
```

bmartel committed:
Update label_studio/frontend/src/components/DraftGuard/DraftGuard.js

Travis1282 committed:
add feature flag

Travis1282 committed:
Merge branch 'fb-optic-2/save-draft-on-leave' of https://github.com/heartexlabs/label-studio into fb-optic-2/save-draft-on-leave

Travis1282 committed:
[submodules] Build static HumanSignal/dm2

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6099106433

Travis1282 committed:
Merge branch 'origin/develop' into 'fb-optic-2/save-draft-on-leave'

Travis1282 committed:
[submodules] Build static HumanSignal/dm2

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6099123242

Travis1282 committed:
add toast to lso

Travis1282 committed:
Merge branch 'fb-optic-2/save-draft-on-leave' of https://github.com/heartexlabs/label-studio into fb-optic-2/save-draft-on-leave

Travis1282 committed:
message toast import fix

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6114309463

Travis1282 committed:
[submodules] Build static HumanSignal/dm2

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6114412393

Travis1282 committed:
Merge branch 'origin/develop' into 'fb-optic-2/save-draft-on-leave'

Travis1282 committed:
[submodules] Build static HumanSignal/dm2

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6114441739

Travis1282 committed:
[submodules] Build static HumanSignal/dm2

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6150533961

Travis1282 committed:
[submodules] Build static HumanSignal/dm2

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6150877518

Travis1282 committed:
fix feature flag

Travis1282 committed:
Merge branch 'fb-optic-2/save-draft-on-leave' of https://github.com/heartexlabs/label-studio into fb-optic-2/save-draft-on-leave

Travis1282 committed:
[submodules] Build static HumanSignal/dm2

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6161293100

Travis1282 committed:
[submodules] Build static HumanSignal/dm2

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6161303530

Travis1282 committed:
[submodules] Build static HumanSignal/dm2

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6164632430

Travis1282 committed:
[submodules] Build static HumanSignal/label-studio-frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6164636576

Travis1282 committed:
[submodules] Build static HumanSignal/dm2

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6164750707

bmartel committed:
[submodules] Build static HumanSignal/dm2

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6173795893

Travis1282 committed:
[submodules] Build static HumanSignal/label-studio-frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6177525481

robot-ci-heartex commented:
To enable Auto Merge for this PR also merge those PRs:
- https://github.com/HumanSignal/dm2/pull/231

Travis1282 committed:
[submodules] Build static HumanSignal/dm2

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6177595440

bmartel commented:
/git merge develop

bmartel committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-2/save-draft-on-leave

bmartel commented:
/frontend help
> Command | Description
> --- | ---
> /frontend build           | Get build frontend static for all repos
> /frontend build lsf [sha] | Get build frontend static for HumanSignal/label-studio-frontend only
> /frontend build dm [sha]  | Get build frontend static for HumanSignal/dm2 only

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6278172019

bmartel committed:
fix the saving of drafts when a user would leave the labelling screen

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6278960962

bmartel committed:
remove console logs

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/6279041796

bmartel commented:
Closing in favor of #4815
